app/src/lib/download.ts

```diff
@@ -2,19 +2,33 @@ import { getAuthHeader } from "./auth/getAuthHeader";
 import appConfig from "../config/appConfig";
 import { FileMetadata } from "../types";
 
-export const getZipUrl = (packetId: string, files: FileMetadata[]) => {
-  const paths = files.map((file) => file.path);
+const filesToPathsParam = (files: FileMetadata[]) =>
+  files.map((file) => `paths=${encodeURIComponent(file.path)}`).join("&");
 
-  return `${appConfig.apiUrl()}/packets/${packetId}/zip?paths=${encodeURIComponent(paths.join(","))}`;
-};
+const filesUrl = (packetId: string, files: FileMetadata[], token: string, filename: string, inline = false) =>
+  `${appConfig.apiUrl()}/packets/${packetId}/files?` +
+  `${filesToPathsParam(files)}&token=${token}&filename=${filename}&inline=${inline}`;
```

Comment:
This is going to misbehave if `filename` has some odd characters.

We should be using some proper API to do this really, not some custom string interpolation.

```js
const url = new URL(`${appConfig.apiUrl()}/packets/${packetId}/files`);
url.searchParams.append('filename', filename);
for (const p of files) {
  url.searchParams.append('paths', p.path);
}
```

[this seems to be a thing](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append)

This should remove the need for `encodeURIComponent`.

Reply:
Great, I'm now using this API.